app/assets/javascripts/app/app.js

```diff
@@ -155,6 +156,29 @@ var app = {
 
   _changeLocation: function(href) {
     window.location.assign(href);
+  },
+
+  setupReport: function() {
+    document.getElementById("report-content-form").addEventListener("submit", function(ev) {
```

Comment:
So I changed the way we deal with the problem, the event is attached only once when the page is loaded, and the data about the post/comment is added to the form when the user click on the icon to open the modal. This ensures no multiple events, and that it's always the good post which is reported. Much cleaner.